src/java-runtime/java/mono/android/Runtime.java

```diff
@@ -14,7 +14,7 @@ private Runtime ()
 	public static native void init (String lang, String[] runtimeApks, String runtimeDataDir, String[] appDirs, ClassLoader loader, String[] externalStorageDirs, String[] assemblies, String packageName, int apiLevel, String[] environmentVariables);
 	public static native void register (String managedType, java.lang.Class nativeClass, String methods);
 	public static native void notifyTimeZoneChanged ();
-	public static native int createNewContext (String[] runtimeApks, String[] assemblies, ClassLoader loader);
+	public static native int createNewContext (String[] runtimeApks, String[] assemblies, byte[][] assembliesBytes, ClassLoader loader);
```

Comment:
I think this would be a good juncture to start introducing aspects of https://github.com/xamarin/designer/issues/1597

In particular, instead of adding these methods to `Runtime.java`, could we please instead introduce a new `DesignerRuntime` type for these Designer-specific methods?  That will make it easier for us to reason about whether a particular change is more or less likely to impact the Designer.  (It is *very* easy for us to "overlook" `Runtime.init()` and inadvertently break the designer.)

Reply:
Could/should there be a `loadAssemblyWithinContext()` method?  Is there a reason that (presumably) "all" assemblies need to be loaded at context creation time?

If it *isn't* necessary to load all the assemblies at once, or if they could be loaded "soon after" context creation time, then you could presumably maintain a "backward-compatible" API interface that involves the existing `Runtime.createNewContext()` + `DesignerRuntime.loadAssemblies(ClassLoader, String[], byte[][])` invocations.  This might be slightly slower, but means there won't be a nasty ABI break to deal with either.

Reply:
I moved the API break to a new method variant so that it's still retrocompatible with previous designer versions (as shown by the test integrations). I would vote for doing the full move to a `DesignerRuntime` class in a later PR.